### DISCUSSION TITLE
Query the current text language / Get access to a default parameter value

### DISCUSSION THREAD

**Discussion by edgarogh** [typst/typst#1793] [Q&A] [answered]
Hello, I'm making a template that contains small bits of text that I would like to translate in two languages (the template is for a formal document that wouldn't make sense in another language anyway).

For simplicity, I would like to rely on the `#text(lang: )` parameter for this. Ideally, there would be a way to retrieve the default value of this parameter (including subsequent `#set` overrides, which would make my code call site-sensitive, but that's fine for me).

Is there such a feature in Typst? I have tried playing with `query` but it doesn't find any `text`, and `text("").at("lang", default: "en")` doesn't return anything else than `"en"`.

**Comment by laurmaedje** [comment, marked as answer]
Unfortunately, this feature doesn't exist yet, but it is planned. It's tracked at this issue: #763.
